ExaFunctions/DurableCsvBulkImportFromBlobStorage.cs

```diff
@@ -0,0 +1,184 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Exasol.Utilities;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Extensions.Logging;
+using MoreLinq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Exasol
+{
+    //http://dontcodetired.com/blog/post/Understanding-Azure-Durable-Functions-Part-4-Passing-Input-To-Orchestrations-and-Activities
+    //https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-create-first-csharp?pivots=code-editor-visualstudio
+    //https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-error-handling?tabs=csharp
+    public static class DurableCsvBulkImportFromBlobStorage
+    {
+        [FunctionName("DurableCsvBulkImportFromBlobStorage")]
+        public static async Task<List<string>> RunOrchestrator(
+            [OrchestrationTrigger] IDurableOrchestrationContext context)
+        {
+
+            ImportOrchestratorParams iop = context.GetInput<ImportOrchestratorParams>();
+            //get the filenames
+            var filenamesList = await context.CallActivityAsync<List<string>>("DurableCsvBulkImportFromBlobStorage_GetFileNames", iop);
+            //divide them in batches 
+            var filenameBatches = filenamesList.Batch(iop.FilesProcessedInParallel.Value).ToList();
+            //https://stackoverflow.com/questions/11463734/split-a-list-into-smaller-lists-of-n-size
+            var outputs = new List<string>();
+            int batchNr = 1;
+            foreach (var filenamesBatch in filenameBatches)
+            {
+                var bulkImportParams = new BulkImportParams { IoParams = iop, FileNames = filenamesBatch.ToList(), BatchNr = batchNr };
+                var retValue = await context.CallActivityAsync<string>("DurableCsvBulkImportFromBlobStorage_RunBulkImporter", bulkImportParams);
+                outputs.Add(retValue);
+                batchNr++;
+            }
+
+            return outputs;
+        }
+        public class BulkImportParams
```

Comment:
Would be nice to have an empty line between classes and methods

Reply:
done